imageio/core/imopen.py

```diff
@@ -1,23 +1,109 @@
-import numpy as np
+from pathlib import Path
 
-from .format import FormatManager, MODENAMES
-from .request import IOMode, Request, InitializationError
+from .request import Request, InitializationError
+from ..config.plugins import PluginConfig
+from ..config import known_plugins
+from ..config.extensions import known_extensions
 
 
-class imopen:
-    """Open a URI and return a plugin instance that can read/write the URIs content.
+def _get_config(plugin: str, legacy_mode: bool) -> PluginConfig:
+    """Look up the config for the given plugin name
 
-    ``imopen`` takes a URI and searches for a plugin capable of opening it. Once
-    a suitable plugin is found, a plugin instance is created that implements the
-    imageio API to interact with the image data. The search can be skipped using
-    the optional ``plugin="plugin_name"`` argument.
+    Factored out for legacy compatibility with FormatManager. Move
+    back into imopen in V3.
+    """
+
+    # for v2 compatibility, delete in v3
+    extension_name = None
+
+    if legacy_mode and Path(plugin).suffix.lower() in known_extensions:
+        # for v2 compatibility, delete in v3
+        extension_name = Path(plugin).suffix.lower()
+    elif plugin in known_plugins:
+        pass
+    elif not legacy_mode:
+        raise ValueError(f"`{plugin}` is not a registered plugin name.")
+    elif plugin.upper() in known_plugins:
+        # for v2 compatibility, delete in v3
+        plugin = plugin.upper()
+    elif plugin.lower() in known_extensions:
+        # for v2 compatibility, delete in v3
+        extension_name = plugin.lower()
+    elif "." + plugin.lower() in known_extensions:
+        # for v2 compatibility, delete in v3
+        extension_name = "." + plugin.lower()
+    else:
+        # for v2 compatibility, delete in v3
+        raise IndexError(f"No format known by name `{plugin}`.")
+
+    # for v2 compatibility, delete in v3
+    if extension_name is not None:
+        for plugin_name in [
+            x
+            for file_extension in known_extensions[extension_name]
+            for x in file_extension.priority
+        ]:
+            if known_plugins[plugin_name].is_legacy:
+                plugin = plugin_name
+                break
+        else:  # pragma: no cover
+            # currently there is no format that is only supported
+            # by v3 plugins
+            raise IndexError(f"No format known by name `{plugin}`.")
+
+    return known_plugins[plugin]
+
+
+def imopen(
+    uri,
+    io_mode: str,
+    *,
+    plugin: str = None,
+    legacy_mode: bool = True,
+    **kwargs,
+):
```

Comment:
Really happy to see this being a function ;)

Reply:
@almarklein I will move imopen into `v3_api.py` in the future, because it seems to belong there now. With this change, we will have a purely functional user-facing API, and I think there is no need for an extra file for `imopen`.